WalletWasabi.Fluent/Views/MainWindow.axaml.cs

```diff
@@ -19,4 +21,22 @@ private void InitializeComponent()
 		this.AttachDevTools();
 #endif
 	}
+
+	protected override void OnClosing(CancelEventArgs e)
+	{
+		base.OnClosing(e);
+
+		if (Application.Current?.DataContext is ApplicationViewModel avm)
+		{
+			if (!Services.UiConfig.HideOnClose)
+			{
+				e.Cancel = !avm.CanShutdown();
+
+				if (e.Cancel)
+				{
+					avm.OnClosePrevented();
+				}
+			}
+		}
+	}
```

Comment:
Couldn't this be moved to `App.axaml.cs`'s `CreateMainWindow()` method rather than code behind?